service/build.go

```diff
@@ -69,18 +76,19 @@ func (uis *UIServer) buildPage(w http.ResponseWriter, r *http.Request) {
 		buildAsUI.Repo = projCtx.ProjectRef.Repo
 	}
 
-	tasks, err := task.FindAllFirstExecution(task.ByBuildId(projCtx.Build.Id))
+	uiTasks, err := getUiTaskCache(projCtx.Build)
 	if err != nil {
 		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "can't get tasks for build"))
 		return
 	}
-	uiTasks, err := getUiTaskCache(projCtx.Build, tasks)
+	buildAsUI.Tasks = uiTasks
+
+	timeTaken, makespan, err := projCtx.Build.GetTimeSpent()
 	if err != nil {
-		uis.LoggedError(w, r, http.StatusInternalServerError, err)
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "can't get time spent for build"))
 		return
 	}
-	buildAsUI.Tasks = uiTasks
-	buildAsUI.TimeTaken, buildAsUI.Makespan = task.GetTimeSpent(tasks)
+	buildAsUI.TimeTaken, buildAsUI.Makespan = timeTaken, makespan
```

Comment:
since err is defined at 79 already, you could just assign this directly at 86 and use =

Reply:
done